src/main/java/com/aws/greengrass/logmanager/util/PeriodicBuffer.java

```diff
@@ -0,0 +1,166 @@
+ */
+public class PeriodicBuffer<K, V> {
+    private static final Logger logger = LogManager.getLogger(PeriodicBuffer.class);
+
+    private final Map<K, V> buffer = new ConcurrentHashMap<>();
+    private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+    private final Consumer<Map<K, V>> flushHandler;
+    private final long intervalSeconds;
+    private final String bufferName;
+    private final Object bufferLock = new Object();
+
+    private ScheduledFuture<?> flushTask;
+    private volatile boolean isShutdown = false;
+
+    /**
+     * Creates a new periodic buffer.
+     *
+     * @param bufferName Name for logging purposes
+     * @param intervalSeconds How often to flush the buffer
+     * @param flushHandler Function to handle flushing buffered items
+     */
+    public PeriodicBuffer(String bufferName, long intervalSeconds, Consumer<Map<K, V>> flushHandler) {
+        this.bufferName = bufferName;
+        this.intervalSeconds = intervalSeconds;
+        this.flushHandler = flushHandler;
+    }
+
+    /**
+     * Starts the periodic flushing.
+     */
+    public void start() {
+        if (isShutdown) {
+            throw new IllegalStateException("Buffer has been shutdown");
+        }
+
+        flushTask = scheduler.scheduleAtFixedRate(
+                this::flush,
+                intervalSeconds,
+                intervalSeconds,
+                TimeUnit.SECONDS
+        );
+
+        logger.atInfo().log("Started periodic buffer '{}' with {}-second interval", bufferName, intervalSeconds);
+    }
+
+    /**
+     * Adds an item to the buffer.
+     *
+     * @param key The key
+     * @param value The value
+     */
+    public void put(K key, V value) {
+        if (isShutdown) {
+            logger.atWarn().log("Attempted to add to shutdown buffer '{}'", bufferName);
+            return;
+        }
+
+        synchronized (bufferLock) {
+            buffer.put(key, value);
+        }
+    }
+
+    /**
+     * Removes an item from the buffer.
+     *
+     * @param key The key to remove
+     */
+    public void remove(K key) {
+        synchronized (bufferLock) {
+            buffer.remove(key);
+        }
+    }
+
+    /**
+     * Manually triggers a flush of the buffer.
+     */
+    public void flush() {
+        synchronized (bufferLock) {
+            if (buffer.isEmpty()) {
+                return;
+            }
+
+            logger.atInfo().log("Flushing buffer '{}' with {} items", bufferName, buffer.size());
+
+            try {
+                // Create a copy to pass to the handler
+                Map<K, V> bufferCopy = new ConcurrentHashMap<>(buffer);
+                flushHandler.accept(bufferCopy);
+
+                // Clear the buffer after successful flush
+                buffer.clear();
+
+                logger.atDebug().log("Successfully flushed buffer '{}'", bufferName);
+            } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that your code handles a broad swath of exceptions in the catch block, potentially trapping dissimilar issues or problems that should not be dealt with at this point in the program.

[Learn more](https://cwe.mitre.org/data/definitions/396.html)